vizro-core/src/vizro/actions/_action_loop/_build_action_loop_callbacks.py

```diff
@@ -22,12 +25,22 @@ def _build_action_loop_callbacks() -> None:
 
     gateway_inputs: List[Input] = []
     for actions_chain in actions_chains:
+        actions_chain_trigger_component_id = actions_chain.trigger.component_id
+        try:
+            actions_chain_trigger_component = model_manager[ModelID(str(actions_chain_trigger_component_id))]
+            # Use underlying table object as a trigger component.
+            if isinstance(actions_chain_trigger_component, Table):
+                actions_chain_trigger_component_id = actions_chain_trigger_component._underlying_table_id
+        # Not all action_chain_trigger_components are included in model_manager e.g. on_page_load_action_trigger
+        except KeyError:
+            pass
```

Comment:
How would we abstract this logic in the future? The Table and a future react model essentially need a way to provide the `underlying` id right?

Should we in this case not work already with something more abstract in the table model such as:

`_callable_object_id` and then work here with `hasattr("_callable_object_id)`?

Reply:
Ah, just saw we were reviewing at the same time haha I suggested the same and was proposing `_figure_id`.
`_callable_object_id` would also be fine for me though